game/game/gamescript.cpp

```diff
@@ -1652,8 +1652,8 @@ int GameScript::wld_getmobstate(std::shared_ptr<zenkit::INpc> npcRef, std::strin
     return -1;
     }
 
-  auto mob = npc->detectedMob();
-  if(mob==nullptr || mob->schemeName()!=scheme) {
+  auto mob = world().findMob(*npc,scheme);
```

Comment:
Should this line be a part of  `detectedMob`?

Idea behind `detectedMob` is to return:
* actively used modsi
* mobsi that cause collision event 
* [new] closemost with valid scheme